### PR TITLE
Allow to flush data from writecache

### PR DESCRIPTION
**Pull request by fyrchik**

### CODE REVIEW

**Review by carpawell — COMMENTED** [review]
The last commit msg is not capitalized.

File: pkg/local_object_storage/shard/writecache.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import (
+	"errors"
+
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard/mode"
+)
+
+// ErrWriteCacheDisabled is returned when an operation on write-cache is performed,
+// but write-cache is disabled.
+var ErrWriteCacheDisabled = errors.New("write-cache is disabled")
```
Comment (cthulhu-rider) [inline pkg/local_object_storage/shard/writecache.go:11]: Can we leave it unexported? I don't see where else we use it. Same for `ErrMustBeReadOnly`.

**Review by cthulhu-rider — APPROVED** [review]

File: pkg/local_object_storage/shard/writecache.go (hunk 1 above)
Comment (fyrchik) [reply pkg/local_object_storage/shard/writecache.go:11]: Fixed

File: pkg/local_object_storage/shard/writecache.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+// FlushWriteCache moves writecache in read-only mode and flushes all data from it.
+// After the operation writecache will remain read-only mode.
+func (s *Shard) FlushWriteCache() error {
+	if !s.hasWriteCache() {
+		return ErrWriteCacheDisabled
+	}
+
+	s.m.Lock()
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/local_object_storage/shard/writecache.go:20]: Yes, fixed

File: pkg/local_object_storage/writecache/flush.go [hunk 3]
```diff
@@ -224,3 +229,68 @@ func (c *cache) flushObject(obj *object.Object) error {
 	_, err = c.metabase.Put(pPrm)
 	return err
 }
+
+// Flush flushes all objects from the write-cache to the main storage.
+// Write-cache must be in readonly mode to ensure correctness of an operation and
+// to prevent interference with background flush workers.
+func (c *cache) Flush() error {
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/local_object_storage/writecache/flush.go]: Fixed

**Review by carpawell — DISMISSED** [review]

**Comment by fyrchik** [comment]
@carpawell Only updated changelog

**Review by carpawell — DISMISSED** [review]

**Comment by fyrchik** [comment]
Rebased on master

**Review by carpawell — APPROVED** [review]

**Pull request merged**